docs/website/docs/general-usage/pipeline.md

```diff
@@ -92,6 +92,28 @@ experimenting a lot. If you want that each time the pipeline resets its state an
 new dataset, set the `dev_mode` argument of the `dlt.pipeline` method to True. Each time the
 pipeline is created, `dlt` adds datetime-based suffix to the dataset name.
 
+## Refresh pipeline data and state
+
+You can use the `refresh` argument of `dlt.pipeline` to fully or partially reset the destination
+dataset and pipeline state when running the pipeline. This is useful if you need to reload all
+data from the beginning or remove/re-create tables in the destination.
+
+This involves dropping or wiping data from tables and fully or partially wiping the pipeline state.
+
+There are three possible refresh modes that can be used. The `refresh` argument should have one of these string values:
+
+* `drop_dataset`
+  All tables listed in the pipeline's schema will be dropped and all source state is wiped.
```

Comment:
we drop only the tables for schemas actually processed right? so if we have two schemas (two different sources) in the pipeline and we use drop and load only source_2, only that source will be wiped? if so we should add a test for that

heh maybe we should rename it to `drop_schemas` or `drop_sources`?

Reply:
This was in fact not being done. It was wiping al source state.
Fixed and tested now. `drop_sources` is good naming imo